### DISCUSSION TITLE
Port issue with multiple Reflex Docker containers

### DISCUSSION THREAD

**Discussion by chimeracle** [reflex-dev/reflex#2043] [Q&A] [answered]
Is there anyone using Reflex for multiple small apps and hosting them in docker containers on the same host? How do you handle the issue where each one wants the same ports? Can the backend be changed from 8000? can the frontend be changed from 3000?

**Comment by masenf** [comment, marked as answer]
Check out the config knobs / environment vars `BACKEND_PORT`, `FRONTEND_PORT` and `API_URL`, they should let you change where the front and back are listening. The `API_URL` currently needs to point to the publicly accessible location of the backend, so if you're running in docker, this would have to be your external hostname.

Maybe something like

```Dockerfile
ENV BACKEND_PORT=8001 FRONTEND_PORT=3001 API_URL=http://example.com:8001
CMD reflex run
```
